src/battle_anim.c

```diff
@@ -0,0 +1,1852 @@
+        animBankId = ANIM_BANK_ATK_PARTNER;
+    else if (animBankId == ANIM_BANK_TARGET)
+        animBankId = ANIM_BANK_DEF_PARTNER;
+
+    if (animBankId == ANIM_BANK_ATTACKER || animBankId == ANIM_BANK_ATK_PARTNER)
+        bank = gAnimBankAttacker;
+    else
+        bank = gAnimBankTarget;
+
+    if (IsAnimBankSpriteVisible(bank))
+    {
+        u8 identity = GetBankIdentity(bank);
+        if (identity == IDENTITY_OPPONENT_MON1 || identity == IDENTITY_PLAYER_MON2 || IsContest())
+            toBG_2 = FALSE;
+        else
+            toBG_2 = TRUE;
+
+        sub_80A438C(bank, toBG_2, FALSE);
+    }
+
+    bank ^= BIT_MON;
+    if (animBankId > 1 && IsAnimBankSpriteVisible(bank))
+    {
+        u8 identity = GetBankIdentity(bank);
+        if (identity == IDENTITY_OPPONENT_MON1 || identity == IDENTITY_PLAYER_MON2 || IsContest())
+            toBG_2 = FALSE;
+        else
+            toBG_2 = TRUE;
+
+        sub_80A438C(bank, toBG_2, FALSE);
+    }
+
+    sBattleAnimScriptPtr++;
+}
+
+static void ScriptCmd_clearmonbg_23(void)
+{
+    u8 animBankId;
+    u8 bank;
+    u8 taskId;
+
+    sBattleAnimScriptPtr++;
+    animBankId = sBattleAnimScriptPtr[0];
+
+    if (animBankId == ANIM_BANK_ATTACKER)
+        animBankId = ANIM_BANK_ATK_PARTNER;
+    else if (animBankId == ANIM_BANK_TARGET)
+        animBankId = ANIM_BANK_DEF_PARTNER;
+
+    if (animBankId == ANIM_BANK_ATTACKER || animBankId == ANIM_BANK_ATK_PARTNER)
+        bank = gAnimBankAttacker;
+    else
+        bank = gAnimBankTarget;
+
+    if (IsAnimBankSpriteVisible(bank))
+        gSprites[gBankSpriteIds[bank]].invisible = 0;
+    if (animBankId > 1 && IsAnimBankSpriteVisible(bank ^ BIT_MON))
+        gSprites[gBankSpriteIds[bank ^ BIT_MON]].invisible = 0;
+    else
+        animBankId = 0;
+
+    taskId = CreateTask(sub_80A4BB0, 5);
+    gTasks[taskId].data[0] = animBankId;
+    gTasks[taskId].data[2] = bank;
+
+    sBattleAnimScriptPtr++;
+}
+
+static void sub_80A4BB0(u8 taskId)
+{
+    gTasks[taskId].data[1]++;
+    if (gTasks[taskId].data[1] != 1)
+    {
+        bool8 toBG_2;
+        u8 bank = gTasks[taskId].data[2];
+        u8 identity = GetBankIdentity(bank);
+        if (identity == IDENTITY_OPPONENT_MON1 || identity == IDENTITY_PLAYER_MON2 || IsContest())
+            toBG_2 = FALSE;
+        else
+            toBG_2 = TRUE;
+
+        if (IsAnimBankSpriteVisible(bank))
+            sub_80A477C(toBG_2);
+        if (gTasks[taskId].data[0] > 1 && IsAnimBankSpriteVisible(bank ^ BIT_MON))
+            sub_80A477C(toBG_2 ^ 1);
+
+        DestroyTask(taskId);
+    }
+}
+
+#undef t1_MONBG_BANK
+#undef t1_MON_IN_BG2
+#undef t1_CREATE_ANOTHER_TASK
+#undef t1_IS_SECONDMON_BG
+
+#undef t2_BANK_SPRITE_ID
+#undef t2_MON_IN_BG2
+#undef t2_MONBG_BANK
+
+static void ScriptCmd_setalpha(void)
+{
+    u16 half1, half2;
+
+    sBattleAnimScriptPtr++;
+    half1 = *(sBattleAnimScriptPtr++);
+    half2 = *(sBattleAnimScriptPtr++) << 8;
+    SetGpuReg(REG_OFFSET_BLDCNT, 0x3F40);
```

Comment:
Consider adding some definition to io_reg.h so you're not just setting an opaque value.